x/protocolpool/keeper/abci.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"context"
+	"time"
+
+	"cosmossdk.io/x/protocolpool/types"
+
+	"github.com/cosmos/cosmos-sdk/telemetry"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k *Keeper) EndBlocker(ctx context.Context) error {
+	defer telemetry.ModuleMeasureSince(types.ModuleName, time.Now(), telemetry.MetricKeyEndBlocker)
+
+	logger := k.Logger(ctx)
+
+	// Iterate over all continuous fund proposals and perform continuous distribution
+	// Note:  This loop intentionally processes all payment streams, and the number of streams
+	// can impact block processing time. However, since it is governed by governance, it is not
+	// considered a denial-of-service (DoS) factor.
+	err := k.ContinuousFund.Walk(ctx, nil, func(key sdk.AccAddress, value types.ContinuousFund) (bool, error) {
```

Comment:
What do we deem to be worst case performance characteristics of this method @likhita-809? That should really drive the epoch discussion.

Reply:
I can think of a few worst case scenarios:
- If there are a large no. of continuous fund proposals, the iteration through all proposals could become time-consuming.
- When there are periods of high network congestion/increased transactions, block processing times may be affected. 
- If the gas cost associated with executing the `continuousDistribution` logic is high, it may lead to increased transaction fees. And also if the gas limit per block is exceeded due to extensive computations or database operations, the block may become invalid because txs could fail to execute, and the block may not be committed.
I think what marko suggests is that we don't do automatic fund distribute which minimizes these kind of factors and adopt lazy fashion